src/os/bluestore/BlueStore.cc

```diff
@@ -4283,9 +4283,11 @@ int BlueStore::_balance_bluefs_freespace(PExtentVector *extents)
     int r = alloc->reserve(gift);
     assert(r == 0);
 
+    thread_local uint64_t hint = 0;
```

Comment:
If we hint here at all, we should use a location from bluefs_extents as the hint.  Probably.. the end of the first extent in the set?